compiler/src/dmd/parse.d

```diff
@@ -8428,7 +8428,12 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                 AST.TemplateParameters* tpl = null;
 
                 nextToken();
-                if (token.value == TOK.leftParenthesis)
+                if (token.value != TOK.leftParenthesis)
```

Comment:
Why is the if-statement being inverted?

Reply:
Because it's better to handle the short case first. Otherwise you have to scroll back for pages to find what the error message clause applies to.